server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1809,16 +2126,32 @@ public static ClusterState updateWithSnapshots(ClusterState state,
      * @param snapshots    snapshots in progress
      * @return shard ids that currently have an actively executing shard snapshot on a data node
      */
-    private static Set<ShardId> busyShardsForRepo(String repoName, @Nullable SnapshotsInProgress snapshots) {
+    private static Set<ShardId> busyShardsForRepo(String repoName, @Nullable SnapshotsInProgress snapshots, Metadata metadata) {
```

Comment:
Maybe `activeShardsForRepo` would be more appropriate?

Reply:
Maybe I'm indifferent to the naming here, I'd say let's not rename things we don't have to rename in this one to keep the noise down maybe? There's a TODO to refactor the use of this method anyway and we can do better with the naming when we get to that?